base/main_test_bucket_pool.go

```diff
@@ -0,0 +1,650 @@
+package base
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/couchbase/gocb"
+	"github.com/couchbaselabs/walrus"
+	"github.com/pkg/errors"
+)
+
+// TestBucketPool is a global instance of a GocbTestBucketPool used to manage a pool of buckets for integration testing.
+var TestBucketPool *GocbTestBucketPool
+
+const (
+	tbpClusterUsername = DefaultCouchbaseAdministrator
+	tbpClusterPassword = DefaultCouchbasePassword
+
+	tbpBucketNamePrefix = "sg_int_"
+
+	// Creates this many buckets in the backing store to be pooled for testing.
+	defaultBucketPoolSize = 10
```

Comment:
Setting this to 10 means it requires a clean cluster to be run (without setting SG_TEST_BUCKET_POOL_SIZE).  I'm wondering whether the default should be either 1 or 5 instead, for ease of use while preserving local dev buckets - what do you think?

Reply:
I'm running some tests now to figure out at what point the test suite starts to slow down.

It looks like running with a pool size of 3 is still only takes 8 minutes, so at worst we could drop the default pool size down to that.

Reply:
A pool size of 1 takes over an to run, a pool size of 2 brings that down to ~9 minutes, and 3+ is ~8 minutes.

I'm going to stick with a default pool size of 3, given we expect to be using multiple buckets per test for sg-replicate.